# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/TransportSecurityInfoUtils.h
# Commit: 23b864e14db0
# Full Hash: 23b864e14db0cf874d0e126b6299c8c44922dde3
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-11-16 09:31:06
# Regressor Bug: 1791633
# File Overlap Count: 1
# Description:
#   Bug 1791633 - separate nsITLSSocketControl from nsITransportSecurityInfo r=necko-reviewers,kershaw,jschanck
#   
#   Depends on D160311
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160313
# ==============================================================================

diff -r 0bcba3375ec0 -r 23b864e14db0 ipc/glue/TransportSecurityInfoUtils.h
--- a/ipc/glue/TransportSecurityInfoUtils.h	Tue Nov 15 20:03:29 2022 +0000
+++ b/ipc/glue/TransportSecurityInfoUtils.h	Tue Nov 15 20:03:29 2022 +0000
@@ -5,6 +5,7 @@
 #ifndef mozilla_ipc_TransportSecurityInfoUtils_h
 #define mozilla_ipc_TransportSecurityInfoUtils_h
 
+#include "ipc/EnumSerializer.h"
 #include "mozilla/RefPtr.h"
 #include "nsITransportSecurityInfo.h"
 #include "nsIX509Cert.h"
@@ -30,6 +31,13 @@
   static bool Read(MessageReader* aReader, RefPtr<nsIX509Cert>* aResult);
 };
 
+template <>
+struct ParamTraits<nsITransportSecurityInfo::OverridableErrorCategory>
+    : public ContiguousEnumSerializerInclusive<
+          nsITransportSecurityInfo::OverridableErrorCategory,
+          nsITransportSecurityInfo::OverridableErrorCategory::ERROR_UNSET,
+          nsITransportSecurityInfo::OverridableErrorCategory::ERROR_TIME> {};
+
 }  // namespace IPC
 
 #endif  // mozilla_ipc_TransportSecurityInfoUtils_h